### PR TITLE
Integrated Green's Function Poisson Solver using heFFTe

### PR DESCRIPTION
**Pull request by Haavaan**
This PR adds distributed FFTs in the Integrated Green's Function (IGF) Poisson Solver, which enhances its computational efficiency. It is useful for improving the performance of the IGF Poisson Solver.

### CODE REVIEW

File: Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp [hunk 1]
```diff
@@ -4,6 +4,8 @@
  *
  * License: BSD-3-Clause-LBNL
  */
+#include <heffte.h>
```
Comment (ax3l) [inline Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp]: Please place between AMReX and stdlibs :)

Comment (RemiLehe) [inline Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp]:
```suggestion
#if defined(ABLASTR_FFT) && defined(ABLASTR_HEFFTE)
#include <heffte.h>
#endif
```

File: Examples/Tests/open_bc_poisson_solver/CMakeLists.txt [hunk 2]
```diff
@@ -11,4 +11,15 @@ if(WarpX_FFT)
...
+        add_warpx_test(
+            test_3d_open_bc_poisson_solver_heffte # name
+            3  # dims
+            2  # nprocs
+            inputs_test_3d_open_bc_poisson_solver_heffte  # inputs
+            analysis.py  # analysis
+            diags/diag1000001  # output
+            OFF  # dependency
+        )
+    endif()
```
Comment (RemiLehe) [inline Examples/Tests/open_bc_poisson_solver/CMakeLists.txt]:
```suggestion
endif()

if(WarpX_HEFFTE)
    add_warpx_test(
        test_3d_open_bc_poisson_solver_heffte # name
        3  # dims
        2  # nprocs
        inputs_test_3d_open_bc_poisson_solver_heffte  # inputs
        analysis.py  # analysis
        diags/diag1000001  # output
        OFF  # dependency
    )
```

File: Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp [hunk 3]
```diff
@@ -26,20 +26,224 @@
 #include <AMReX_MLLinOp.H>
 #include <AMReX_MultiFab.H>
 #include <AMReX_REAL.H>
+#include <AMReX_PlotFileUtil.H>
```
Comment (RemiLehe) [inline Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp]:
```suggestion
```

File: Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp [hunk 4]
```diff
@@ -26,20 +26,224 @@
...
 
-#include <array>
+#if defined(ABLASTR_USE_FFT) && defined(ABLASTR_USE_HEFFTE)
+#include <heffte.h>
+#endif
 
+#include <array>
+#include <fstream>
+#include <iomanip>
```
Comment (RemiLehe) [inline Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp]:
```suggestion
```

File: Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp [hunk 5]
```diff
@@ -36,10 +38,208 @@ void
...
                 amrex::MultiFab & phi,
                 std::array<amrex::Real, 3> const & cell_size,
-                amrex::BoxArray const & ba )
+                amrex::BoxArray const & ba)
 {
     using namespace amrex::literals;
 
+    BL_PROFILE_VAR_NS("IGF FFTs", timer_ffts);
+    BL_PROFILE_VAR_NS("IGF FFT plans", timer_plans);
+    BL_PROFILE_VAR_NS("IGF parallel copies", timer_pcopies);
```
Comment (RemiLehe) [inline Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp]:
```suggestion
    BL_PROFILE_VAR_NS("ablastr::fields::computePhiIGF: FFTs", timer_ffts);
    BL_PROFILE_VAR_NS("ablastr::fields::computePhiIGF: FFT plans", timer_plans);
    BL_PROFILE_VAR_NS("ablastr::fields::computePhiIGF: parallel copies", timer_pcopies);
```

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**